### DISCUSSION TITLE
Shelly 1 as button is slow

### DISCUSSION THREAD

**Discussion by dejan-kosak** [RavenSystem/esp-homekit-devices#2132] [General]
Hi All,

I have Shelly 1 configured as button using latest firmware. 
I am observing very slow response when button is pushed. It takes 3-5s for action to complete. 
In the past (older firmware) action took much less to complete, around 1-2s.

I am not sure what is causing this slow operation, everything else in my home setup is the same.

This is my shelly1 configuration:
`{"c":{"io":[[[4],2],[[5],6,0,1]]},"a":[{"t":3,"0":{"r":[[4]]},"1":{"r":[[4,1]]},"f0":[{"g":5,"t":1}],"f1":[{"g":5,"t":2}],"f2":[{"g":5,"t":3}]}]}`

Any clue?

**Comment by Tellicious** [comment]
Are you sure that input needs to be inverted? otherwise try this:
`{"c":{"io":[[[4],2],[[5],6]]},"a":[{"t":3,"0":{"r":[[4]]},"1":{"r":[[4,1]]},"f0":[[5]],"f1":[[5,2]],"f2":[[5,3]]}]}`

**Reply by dejan-kosak** [reply]
I guess it needs to be inverted, your suggested config does not work properly.

At the end I found out a issue was on my 2,4 network, which was for some unknown reason very slow. I restarted router and it got better.
